include/nbl/video/IAPIConnection.h

```diff
@@ -3,6 +3,7 @@
 
 #include "nbl/core/IReferenceCounted.h"
 #include "nbl/video/IPhysicalDevice.h"
+#include "nbl/video/EApiType.h"
```

Comment:
why move out the api type into a global enum?

Reply:
because probably all of backend classes will return this with some getApiType () and they shouldn't have to include iapiconnection just for this

Reply:
this https://github.com/Devsh-Graphics-Programming/Nabla/pull/28#discussion_r569436172

Reply:
I'm fine including IAPIConnection just for that....

also the backend classes should have a `castDown` method so the process of testing the APIType, debug breaking, returning nullptr/static_cast<CBackendType*>(this) of correct constness is just 1 line, not the same 5 lines of s**t duplicated everywhere